### PR TITLE
Create input range component

### PR DESCRIPTION
**Pull request by SergeyDenisovich**

### CODE REVIEW

File: src/App.tsx [hunk 1]
```diff
@@ -1,8 +1,11 @@
 import React from 'react';
```
Comment (Grishanov-Dmitry) [inline src/App.tsx:1]: Needs add indent between node-modules and src imports

File: src/components/formComponents/InputRangeComponent.tsx [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+export interface InputRangeComponentProps {
+    queryAmount?: number
+}
+
+export const InputRangeComponent: React.FC<InputRangeComponentProps> = ({queryAmount}) => {
+
+    const [range, setRange] = React.useState(queryAmount || 0);
+    const [number, setNumber] = React.useState(queryAmount || 0);
+
+    const handleInputChange = (e: React.ChangeEvent<HTMLInputElement>) => {
```
Comment (Grishanov-Dmitry) [inline src/components/formComponents/InputRangeComponent.tsx:14]: You can do destructurization here
const handleInputChange = ({target} ...

File: src/components/formComponents/InputRangeComponent.tsx [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+        if(e.target.name === 'number') {
+            if(!checkNumber(Number(e.target.value))) return false;
```
Comment (Grishanov-Dmitry) [inline src/components/formComponents/InputRangeComponent.tsx:16]: Don't compare with string. Use constants

File: src/components/formComponents/InputRangeComponent.tsx [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+        }
+
+        setRange(Number(e.target.value));
+        setNumber(Number(e.target.value));
+    }
+
+    // check if the input[number] value in range - 0 - 50
+    const checkNumber = (num: number) => {
+        return num >= 0 && num <= 50;
+    }
```
Comment (Grishanov-Dmitry) [inline src/components/formComponents/InputRangeComponent.tsx:26]: You can use more simple syntactic: 
    const checkNumber = (num: number) => num >= 0 && num <= 50;

File: src/components/formComponents/InputRangeComponent.tsx [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+
+    return (
+        <div className={classes.input_range_block}>
+            <p>
+                <span>Максимальное количество:</span>
+            </p>
+            <div className={classes.input_group}>
```
Comment (Grishanov-Dmitry) [inline src/components/formComponents/InputRangeComponent.tsx:33]: Use camelCase for classes

File: src/index.css [hunk 6]
```diff
@@ -1,5 +1,5 @@
 body {
-  margin: 0;
+  /* margin: 0; */
```
Comment (Grishanov-Dmitry) [inline src/index.css:2]: Delete the commented strings

File: src/components/formComponents/inputRangeComponent/InputRangeComponent.tsx [hunk 7]
```diff
@@ -0,0 +1,43 @@
...
+    queryAmount?: number
+}
+
+export const InputRangeComponent: React.FC<InputRangeComponentProps> = ({queryAmount}) => {
+    // props queryAmount will come to component if user query was saved
+    
+    const [range, setRange] = React.useState(queryAmount || 0);
+    const [number, setNumber] = React.useState(queryAmount || 0);
+
+    const handleInputChange = (e: React.ChangeEvent<HTMLInputElement>) => {
```
Comment (Grishanov-Dmitry) [inline src/components/formComponents/inputRangeComponent/InputRangeComponent.tsx:17]: Distructurization!

File: src/components/formComponents/inputRangeComponent/InputRangeComponent.tsx [hunk 8]
```diff
@@ -0,0 +1,43 @@
...
+import {InputRangeComponent_MIN_MAX_QUERY_AMOUNT} from '../../../constants/constants';
+
+import styles from './InputRangeComponent.module.css';
+
+export interface InputRangeComponentProps {
+    queryAmount?: number
+}
+
+export const InputRangeComponent: React.FC<InputRangeComponentProps> = ({queryAmount}) => {
+    // props queryAmount will come to component if user query was saved
```
Comment (Grishanov-Dmitry) [inline src/components/formComponents/inputRangeComponent/InputRangeComponent.tsx:12]: Delete coment

File: src/components/formComponents/inputRangeComponent/InputRangeComponent.tsx [hunk 9]
```diff
@@ -0,0 +1,43 @@
...
+    const handleInputChange = (e: React.ChangeEvent<HTMLInputElement>) => {
+        if(e.target.name === 'numberInput') {
+            if(!checkNumber(Number(e.target.value))) return false;
+        }
+
+        setRange(Number(e.target.value));
+        setNumber(Number(e.target.value));
+    }
+
+    // limit query
```
Comment (Grishanov-Dmitry) [inline src/components/formComponents/inputRangeComponent/InputRangeComponent.tsx:26]: Move to the helpers folder if you will use it in different place

Comment (SergeyDenisovich) [reply src/components/formComponents/inputRangeComponent/InputRangeComponent.tsx:26]: This check func will attend only in this component

**Review by Grishanov-Dmitry — APPROVED** [review]

**Review by Grishanov-Dmitry — APPROVED** [review]

**Pull request merged**